### PR TITLE
Http2 noalloc header compare 5454 v3

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/5454

Describe changes:
- http2: remove `to_vec` in comparisons (no need to alloc)

Replaces #7671 with additional commit to fix another clippy warning

### CODE REVIEW

File: rust/src/http2/decompression.rs [hunk 1]
```diff
@@ -165,16 +165,16 @@ impl HTTP2DecoderHalf {
         }
     }
 
-    pub fn http2_encoding_fromvec(&mut self, input: &Vec<u8>) {
+    pub fn http2_encoding_fromvec(&mut self, input: &[u8]) {
         //use first encoding...
         if self.encoding == HTTP2ContentEncoding::HTTP2ContentEncodingUnknown {
-            if *input == "gzip".as_bytes().to_vec() {
+            if &input[..] == "gzip".as_bytes() {
```
Comment (jasonish) [inline rust/src/http2/decompression.rs:171]: These can now simply be `input == "gzip".as_bytes()`...  Or even `input == b"gzip"`.

Comment (catenacyber) [reply rust/src/http2/decompression.rs:171]: `input == "gzip".as_bytes()` does not work on rust 1.48 :-/

Comment (catenacyber) [reply rust/src/http2/decompression.rs:171]: Or do you mean I should invert the commits order..? Testing...

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/7692

**Pull request closed without merging**